src/pytest_servers/s3.py

```diff
@@ -37,49 +37,29 @@ def __exit__(self, *exc_args):
         self._server.stop()
 
 
-@pytest.fixture(scope="session")
-def s3_fake_creds_file(monkeypatch_session: pytest.MonkeyPatch) -> None:  # type: ignore[misc]
-    # https://github.com/spulec/moto#other-caveats
-    import pathlib
-
-    aws_dir = pathlib.Path("~").expanduser() / ".aws"
-    aws_dir.mkdir(exist_ok=True)
-
-    aws_creds = aws_dir / "credentials"
-    initially_exists = aws_creds.exists()
-
-    if not initially_exists:
-        aws_creds.touch()
-
-    try:
-        with monkeypatch_session.context() as m:
-            try:
-                m.delenv("AWS_PROFILE")
-            except KeyError:
-                pass
-            m.setenv("AWS_ACCESS_KEY_ID", "pytest-servers")
-            m.setenv("AWS_SECRET_ACCESS_KEY", "pytest-servers")
-            m.setenv("AWS_SECURITY_TOKEN", "pytest-servers")
-            m.setenv("AWS_SESSION_TOKEN", "pytest-servers")
-            m.setenv("AWS_DEFAULT_REGION", "us-east-1")
-            yield
-    finally:
-        if aws_creds.exists() and not initially_exists:
-            aws_creds.unlink()
-
-
 @pytest.fixture(scope="session")
 def s3_server_config() -> dict:
-    """Override to change default config of the server."""
+    """Override to change default config of the moto server."""
     return {}
 
 
 @pytest.fixture(scope="session")
 def s3_server(  # type: ignore[misc]
+    monkeypatch_session: pytest.MonkeyPatch,
     s3_server_config: dict,
-    s3_fake_creds_file: None,  # noqa: ARG001
-) -> str:
-    """Spins up a moto s3 server. Returns the endpoint URL."""
+) -> dict[str, str | None]:
+    """Spins up a moto s3 server.
+
+    Returns a client_kwargs dict that can be used with a boto client.
```

Comment:
C: in can be used also to set env var per test (e.g. creating a function scoped fixture). E.g. in my case I don't have access directly to S3 boto. So, I would still need to put them into evn or credentials file

Reply:
... or pass them into DVC config - that might work as well in my case

Reply:
I guess, actionable discussion here / thought is if we want to have an extra fixture, function scoped that sets env vars.... probably it's fine to leave it to the client

Reply:
> E.g. in my case I don't have access directly to S3 boto. So, I would still need to put them into evn or credentials file

Depending on what you're trying to accomplish, you can use the `fs` attribute of the upath instance to access `S3FileSystem` and eventually even use the underlying client (although it's async, so your mileage may vary)

```python

def test_something(tmp_s3_path):
    fsspec_fs = tmp_s3_path.fs
    aiobotocore_client = fsspec_fs
    ...
```

Regarding the fixture to set env vars: maybe we can work something out in another PR, I have some code but it has issues with fixture scopes.

Reply:
There are plenty of tests where we won't have access to boto, etc to pass the config directly I think.

And example of a test is you use dvc repo api to push / pull. The only think you control is dvc remote config or env vars, there is not direct access to boto usually. 

 But I also think it's fine for now. Client can put together a simple fixture to override env, or set DVC config or some other config. It's way more flexible now with these all values available.

I can try to use it after we merge and let you know if that works or not.